File: binary-tree-level-order-traversal/src/test/java/org/example/MainTest.java
package org.example;

import org.junit.jupiter.api.Test;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.InputStream;
import java.io.PrintStream;
import java.nio.charset.StandardCharsets;

class MainTest {

    @Test
    void mainTest() {
        var result = MockTest("3 9 20 null null 15 7 !\n", () -> Main.main(new String[]{}));
        System.out.println(result);
    }

    synchronized private static String MockTest(String input, Runnable runnable) {
        PrintStream orig = System.out;

        ByteArrayOutputStream os = new ByteArrayOutputStream();
        System.setOut(new PrintStream(os, false, StandardCharsets.UTF_8));

        InputStream is = System.in;
        System.setIn(new ByteArrayInputStream(input.getBytes()));
        try {
            runnable.run();
        } finally {
            System.setOut(orig);
            System.setIn(is);

        }
        return (os.toString(StandardCharsets.UTF_8));

    }

}

File: binary-tree-level-order-traversal/src/test/kotlin/org/example/MainTest.kt
package org.example

import com.github.masx200.learn_java_kotlin_programming.utils.MockTest
import org.junit.jupiter.api.Test

internal class MainTest {
    @Test
    fun mainTest() {
        val result = MockTest.MockTest("3 9 20 null null 15 7 !\n") { Main.main(arrayOf()) }
        println(result.contentToString())
    }
}